src/array_slice.cpp

```diff
@@ -333,7 +350,7 @@ static void get_array_slice_chunkloop(fields_chunk *fc, int ichnk, component cgr
                                           frequency));
           if (abs(tr) == 0.0) tr += 4.0; // default inveps == 1
         }
-        fields[i] = (data->snap ? 1.0 : IVEC_LOOP_WEIGHT(s0, s1, e0, e1, 1.0)) * (4.0 * data->ninveps) / tr;
+        fields[i] = (data->snap ? 1.0 : IVEC_LOOP_WEIGHT(s0i, s1i, e0i, e1i, 1.0)) * (4.0 * data->ninveps) / tr;
```

Comment:
Just:
```cxx
fields[i] = IVEC_LOOP_WEIGHT(s0i, s1i, e0i, e1i, 1.0) * (4.0 * data->ninveps) / tr;
```
since the interpolation weights are already `1.0` if `snap==true`.